### ISSUE TITLE
add Table.Header and Table.Footer

### ISSUE THREAD

**Issue by plandem** [ant-design/ant-design#4993]
Right now the only way to set footer/title is to use functions, I suggest to add shorthands for it in react-way like for columns. E.g.:
```jsx
<Table bordered size='middle' rowKey='id' dataSource={users}>
	<Table.Title>
		<Action.Bulk selected={selectedRowKeys.length} onClick={(key) => this.handleOnBulkAction(key)}>
			<Action key="groupLink" icon="download" tooltip={formatMessage(messages.multiArchiveIn)}/>
			<Action key="groupUnlink" icon="upload" tooltip={formatMessage(messages.multiArchiveOut)}/>
			<Action key="block" icon="lock" tooltip={formatMessage(messages.multiBlock)}/>
			<Action key="unblock" icon="unlock" tooltip={formatMessage(messages.multiUnblock)}/>
		</Action.Bulk>
		<Button icon="plus" onClick={this.handleOnCreate}>New User</Button>
	</Table.Title>
	<Table.Column title={formatMessage(messages.model.User.username)} dataIndex='username' className='right' width='15%'/>
	<Table.Column title={formatMessage(messages.model.User.name)} dataIndex='name' width='25%'/>
	<Table.Column title={formatMessage(messages.model.User.phone)} dataIndex='phone' className='right' width='15%'/>
	<Table.Column title={formatMessage(messages.columnGroups)} render={this.renderRecordTags} width='35%'/>
	<Table.Column render={this.renderActionsForUser} className='actions'/>
	<Table.Footer>mega footer</Table.Footer>
</Table>
```

I know that sometimes we need to use **currentPageData**, but for that cases we always can fallback for **callback** - current behavior.

P.S.: Because of this I have to use "HOC" for table:
```jsx
import { Table as AntTable } from 'antd';

const TableHeader = ({ className, children }) => {
	return <div className={className}>{children}</div>
};

TableHeader.propTypes = {
	children: PropTypes.any.isRequired,
};

const TableFooter = ({ className, children }) => {
	return <div className={className}>{children}</div>
};

TableFooter.propTypes = {
	children: PropTypes.any.isRequired,
};

class Table extends React.Component {
	render() {
		const { children, ...restProps } = this.props;
		const tableProps = restProps;

		if(Array.isArray(children)) {
			const columns = [];

			React.Children.forEach(children, (child, i) => {
				if(child.type === AntTable.Column || child.type === AntTable.ColumnGroup) {
					columns.push(child);
				} else if(child.type === TableHeader) {
					tableProps.title = () => child;
				} else if(child.type === TableFooter) {
					tableProps.footer = () => child;
				}
			});

			if(columns.length) {
				tableProps.children = columns;
			}
		}

		return <AntTable {...tableProps}/>
	}
}

Table.propTypes = {...AntTable.propTypes};

export default Table;
```

How ofter do you use **header/footer** feature and use **currentPageData**?! In my cases - never, I always return something like:
```jsx
renderTitle = () => {
    return <div>header content</div>
}
```

but in most cases, it's not like 1 line of code

**Comment by yesmeck** [comment]
The current `Table.Column` implementation is a little tricky, it's not describing a DOM structure but a data structure, which means `Table.Column` can not be wrapped by other components or created from HOC. Before we find a better way to implement `Table.Column`, I would downvote to introduce `Table.Header` and `Table.Footer`.

**Comment by bang88** [comment]
Agree @yesmeck ,  write your own `Table` component extends Ant's `Table` for  fit your needs is recommended. 😄

**Comment by KalpanaPagariya** [comment]
It would really be nice to have this feature in to have Table.Header and Table.Footer
